blockmanager.go

```diff
@@ -360,6 +404,19 @@ func (b *blockManager) NewPeer(sp *ServerPeer) {
 	}
 }
 
+// addNewPeerToList adds the peer to the peers list.
+func (b *blockManager) addNewPeerToList(peers *list.List, sp *ServerPeer) {
+	// Ignore if in the process of shutting down.
+	if atomic.LoadInt32(&b.shutdown) != 0 {
+		return
+	}
+
+	log.Infof("New valid peer %s (%s)", sp, sp.UserAgent())
+
+	// Add the peer as a candidate to sync from.
+	peers.PushBack(sp)
+}
```

Comment:
why is this on the blockManager? and why is it checking for shutdown rather than just doing the thing..

Reply:
so I think this particular change and that of the `removeDonePeerFromList` should be in a different commit. This function is an extrapolation of this: https://github.com/lightninglabs/neutrino/blob/42a196facefac36282e68ddf4a02c9ce4601a0b0/blockmanager.go#L367-L381

It was extrapolated because we only need the block handler to carry on with that when a new peer is added while fetching checkpointed block headers.

Reply:
This extraction is rather arbitrary, I cannot think of a good reason why a function whose stated purpose is to add the peer to the list needs to check for shutdown. I have checked all of the callsites of this function and all of them have redundant shutdown logic, so please remove the shutdown logic from this method and then remove the blockManager receiver argument.

Reply:
Looking further, I understand why you did this, but that doesn't mean we should let it remain. The `handleNewPeerMsg` not taking a message is wrong. We need to fix both of them. I'm aware this precedes your involvement but it should be kept consistent and move in a direction of greater readability